addons/hr_holidays/models/hr_leave.py

```diff
@@ -888,8 +888,9 @@ def _validate_leave_request(self):
         meetings = self.env['calendar.event']
         if meeting_holidays:
             meeting_values_for_user_id = meeting_holidays._prepare_holidays_meeting_values()
-            for user_id, meeting_values in meeting_values_for_user_id.items():
-                meetings += self.env['calendar.event'].with_user(user_id or self.env.uid).with_context(
+            for user, meeting_values in meeting_values_for_user_id.items():
+                meetings += self.env['calendar.event'].with_user(user.id or self.env.uid).with_context(
+                                allowed_company_ids=user.company_ids.ids,
```

Comment:
```suggestion
                                allowed_company_ids=[],
```
not tested but as far as I can tell, if `allowed_company_ids` is falsy in context, we should use `env.user.company_ids.ids`.

this would also have the advantage of working also for the `or self.env.uid` part